MdsSupportingClasses/Collivery.php

```diff
@@ -121,12 +121,17 @@ private function consumeAPI($url, $data, $type, $isAuthenticating = false) {
         try {
             $result = curl_exec($client);
             
-
             if (curl_errno($client)) {
                 $errno = curl_errno($client);
                 $errmsg = curl_error($client);
-                new SoapConnectionException("URL:".$url." RESULT:".$result);
-                new SoapConnectionException("CODE:".$errno." MSG:".$errmsg);
+                curl_close($client);
+
+                throw new CurlConnectionException("ConsumeAPI()", "Error executing request", [
```

Comment:
The `CurlConnectionException::__construct()` has `$message, $functionName, array $data`
You are passing in the function name first here

Reply:
I've updated.